### PR TITLE
Allow expanded DNS configuration

### LINKED ISSUE #5198: Support more DNS search paths
**Description**

kubernetes issue: https://github.com/kubernetes/kubernetes/issues/104352

cri-o does not allow DnsConfig to have more than 6 search paths. However, modern libcs allow more search paths(many are unlimited)

According to https://man7.org/linux/man-pages/man5/resolv.conf.5.html,
```
In glibc 2.25 and earlier, the search list is limited to
six domains with a total of 256 characters.  Since glibc
2.26, the search list is unlimited.
```

**Steps to reproduce the issue:**
1. Launch kubernetes v1.22.0 with enabling the feature gate [ExpandedDNSConfig](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/2595-expanded-dns-config)
2. Create a pod with 7 search paths
```yaml
apiVersion: v1
kind: Pod 
metadata:
  creationTimestamp: null
  labels:
    run: test
  name: test
spec:
  containers:
  - args:
    - sleep
    - "3600"
    image: ubuntu
    name: test
    resources: {}
  dnsConfig:
    searches:
    - 1.com
    - 2.com
    - 3.com
    - 4.com
    - 5.com
    - 6.com
    - 7.com
  dnsPolicy: ClusterFirst
  restartPolicy: Always
status: {}
```

**Describe the results you received:**
```sh
$ kubectl describe pods test
...
Events:
  Type     Reason                  Age               From     Message
  ----     ------                  ----              ----     -------
  Warning  FailedCreatePodSandBox  6s (x5 over 45s)  kubelet  Failed to create pod sandbox: rpc error: code = Unknown desc = DNSOption.Searches has more than 6 domains
```

**Describe the results you expected:**

The pod should run successfully.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `crio --version`:**

```
$ crio --version
WARN[0000] Encountered error detecting link mode of binary:  
WARN[0000] Encountered error detecting link mode of binary:  
crio version 1.22.0
Version:       1.22.0
GitCommit:     78f06f200a544ec3be594d58a37d247f822c1946
GitTreeState:  dirty
BuildDate:     1980-01-01T00:00:00Z
GoVersion:     go1.16.6
Compiler:      gc
Platform:      linux/amd64
Linkmode:      unknown
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

### PR DESCRIPTION
**Pull request by gjkim42**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

This PR makes cri-o allow [ExpandedDNSConfig](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/2595-expanded-dns-config)

#### Which issue(s) this PR fixes:

Fixes #5198 

#### Does this PR introduce a user-facing change?

```release-note
cri-o now does not limit the DNS search paths.
```

### CODE REVIEW

**Review by saschagrunert — APPROVED** [review]
/ok-to-test

Thank you @gjkim42! :pray:

File: pkg/sandbox/fixtures/expanded_resolv.conf [hunk 1]
```diff
@@ -0,0 +1,4 @@
+search 1.com 2.com 3.com 4.com 5.com 6.com 7.com
```
Comment (haircommander) [inline pkg/sandbox/fixtures/expanded_resolv.conf:1]: can we clean this up after the test?

Comment (gjkim42) [reply pkg/sandbox/fixtures/expanded_resolv.conf:1]: I think we've already cleaned the output resolv.conf file up after each test.

https://github.com/cri-o/cri-o/blob/c53e78f45ffde7a3b7a3d93d9c8c9410ca0d4fec/pkg/sandbox/infra_test.go#L42-L43

Am I missing something?

Comment (haircommander) [reply pkg/sandbox/fixtures/expanded_resolv.conf:1]: ah the test as it currently is is not quite right. If we do os.Remove(c.Path) after calling Expect(ParseDNSOptions...), it's possible ParseDNSOptions failed after creating the file but before creating. can you switch the order of the defer and Expect calls, and remove this file from the commit?

Comment (gjkim42) [reply pkg/sandbox/fixtures/expanded_resolv.conf:1]: > can you switch the order of the defer and Expect calls

That is a good point. Done.

> and remove this file from the commit?

I think this file is a test data as fixtures/resolv.conf is.

Should we have to refactor the test to be able to test without these files?
Let me know if that is what you request.

Comment (haircommander) [reply pkg/sandbox/fixtures/expanded_resolv.conf:1]: for some reason I didn't see the fixtures directory when I looked before. We can have the refactor as a follow up (which I'd love if you wanted to take that on :smiley:  )

**Comment by haircommander** [comment]
waiting on the gh action to run through, and I will tag after. Thank you for your work @gjkim42 !

**Pull request merged**